internal/config/configdomain/proposal_stack_lineage_builder_test.go

```diff
@@ -0,0 +1,173 @@
+				URL:          "https://www.github.com/git-town/pull/1",
+			},
+			ForgeType: forgedomain.ForgeTypeCodeberg,
+		}), nil
+	})
+}
+
+func (self *mockConnectorProposalStackLineageBuilder) OpenRepository(_ subshelldomain.Runner) error {
+	return nil
+}
+
+func (self *mockConnectorProposalStackLineageBuilder) SearchProposalFn() Option[func(_ gitdomain.LocalBranchName) (Option[forgedomain.Proposal], error)] {
+	return None[func(_ gitdomain.LocalBranchName) (Option[forgedomain.Proposal], error)]()
+}
+
+func (self *mockConnectorProposalStackLineageBuilder) SquashMergeProposalFn() Option[func(_ int, _ gitdomain.CommitMessage) error] {
+	return None[func(_ int, _ gitdomain.CommitMessage) error]()
+}
+
+func (self *mockConnectorProposalStackLineageBuilder) UpdateProposalBodyFn() Option[func(_ forgedomain.ProposalInterface, _ string) error] {
+	return None[func(_ forgedomain.ProposalInterface, _ string) error]()
+}
+
+func (self *mockConnectorProposalStackLineageBuilder) UpdateProposalSourceFn() Option[func(_ forgedomain.ProposalInterface, _ gitdomain.LocalBranchName) error] {
+	return None[func(_ forgedomain.ProposalInterface, _ gitdomain.LocalBranchName) error]()
+}
+
+func (self *mockConnectorProposalStackLineageBuilder) UpdateProposalTargetFn() Option[func(_ forgedomain.ProposalInterface, _ gitdomain.LocalBranchName) error] {
+	return None[func(_ forgedomain.ProposalInterface, _ gitdomain.LocalBranchName) error]()
+}
+
+func (self *mockConnectorProposalStackLineageBuilder) VerifyConnection() forgedomain.VerifyConnectionResult {
+	return forgedomain.VerifyConnectionResult{
+		AuthenticatedUser:   None[string](),
+		AuthenticationError: nil,
+		AuthorizationError:  nil,
+	}
+}
+
+func TestProposalStackLineageBuilder_CheckLineageAndProposals(t *testing.T) {
+	t.Parallel()
+	// arrange
+	mainBranch := gitdomain.NewLocalBranchName("main")
+	perennialBranch := gitdomain.NewLocalBranchName("development")
+	featureBranch := gitdomain.NewLocalBranchName("git-town/proposal-stack-lineage")
+	lineage := configdomain.NewLineage()
+	lineage.Root(mainBranch)
+	lineage.Set(perennialBranch, mainBranch)
+	lineage.Set(featureBranch, perennialBranch)
+	var connector forgedomain.Connector = &mockConnectorProposalStackLineageBuilder{}
+	args := configdomain.ProposalStackLineageArgs{
+		AfterStackDisplay:        []string{},
+		BeforeStackDisplay:       []string{},
+		Connector:                Some(connector),
+		CurrentBranch:            featureBranch,
+		CurrentBranchIndicator:   ":point_left:",
+		IndentMarker:             "-",
+		Lineage:                  lineage,
+		MainAndPerennialBranches: Some(gitdomain.NewLocalBranchNames(mainBranch.String(), perennialBranch.String())),
+	}
+	expectedStackLineage := ` - main
+   - development
+     - PR https://www.github.com/git-town/pull/1 :point_left:
+`
+
+	// act
+	actual := configdomain.NewProposalStackLineageBuilder(&args)
+
+	// assert
+	must.True(t, actual.IsSome())
+	builder := actual.GetOrDefault()
```

Comment:
Here is how you destructure the optional value properly (this follows the way the Rust language does it):

```suggestion
  builder, hasBuilder := actual.Get()
	must.True(t, hasBuilder)
```

Reply:
https://github.com/git-town/git-town/pull/4871/commits/81fdf11c19abcc427c9ab030f7d9628b16c999a4